src/parser/tests/variable_parser_tests.rs

```diff
@@ -422,3 +422,251 @@ fn var_config_location() {
 
     assert_eq!("main.instance.foo", &src[result.var_config[0].location.to_range().unwrap()]);
 }
+
+#[test]
+fn var_external() {
+    let src = r#"
+    VAR_GLOBAL 
+        arr: ARRAY [0..100] OF INT; 
+    END_VAR
+
+    FUNCTION foo
+    VAR_EXTERNAL 
+        arr : ARRAY [0..100] OF INT;
+    END_VAR
+    END_FUNCTION
+    "#;
+
+    let (result, _) = parse(src);
+
+    insta::assert_debug_snapshot!(result, @r###"
+    CompilationUnit {
+        global_vars: [
+            VariableBlock {
+                variables: [
+                    Variable {
+                        name: "arr",
+                        data_type: DataTypeDefinition {
+                            data_type: ArrayType {
+                                name: None,
+                                bounds: RangeStatement {
+                                    start: LiteralInteger {
+                                        value: 0,
+                                    },
+                                    end: LiteralInteger {
+                                        value: 100,
+                                    },
+                                },
+                                referenced_type: DataTypeReference {
+                                    referenced_type: "INT",
+                                },
+                                is_variable_length: false,
+                            },
+                        },
+                    },
+                ],
+                variable_block_type: Global,
+            },
+        ],
+        var_config: [],
+        units: [
+            POU {
+                name: "foo",
+                variable_blocks: [
+                    VariableBlock {
+                        variables: [
+                            Variable {
+                                name: "arr",
+                                data_type: DataTypeDefinition {
+                                    data_type: ArrayType {
+                                        name: None,
+                                        bounds: RangeStatement {
+                                            start: LiteralInteger {
+                                                value: 0,
+                                            },
+                                            end: LiteralInteger {
+                                                value: 100,
+                                            },
+                                        },
+                                        referenced_type: DataTypeReference {
+                                            referenced_type: "INT",
+                                        },
+                                        is_variable_length: false,
+                                    },
+                                },
+                            },
+                        ],
```

Comment:
Does this have to be a pointer to the actual global variable or are external variables imported as locals by deep-cloning them? Just a thought, doesn't change anything in this PR since we don't support them anyways.